rest/routes/user.py

```diff
@@ -43,20 +42,22 @@ def register():
     editor_request = request.json['editorRequest']
 
     data = {}
-    if len(name) < 3 or len(name) > 30 or re.match('^[.a-zA-Z0-9_-]+$', name) is None:
+    if len(name) < 3 or len(name) > 30 or \
+            re.match(r'^[ąĄćĆęĘłŁńŃóÓśŚźŹżŻ.a-zA-Z0-9 _-]+$', name) is None:
         data['name'] = name
-    if not email.endswith('agh.edu.pl') or re.match('^[.@a-zA-Z0-9_-]+$', email) is None:
+    if not email.endswith('agh.edu.pl') or re.match(r'^[.@a-zA-Z0-9_-]+$', email) is None:
         data['email'] = email
-    if len(password) < 3 or len(password) > 30 or re.match('^[.a-zA-Z0-9_-]+$', password) is None:
+    if len(password) < 3 or len(password) > 30 or re.match(r'^[.a-zA-Z0-9_-]+$', password) is None:
         data['password'] = password
 
     if len(data) > 0:
+        data['error'] = 'Invalid registration data'
```

Comment:
You should return a more appropriate error message, I mean when the email format is invalid, the backend ought to return `"Wprowadzony email jest niepoprawny"` or something else, the same applies to name and password.